src/kernels/types.ts

```diff
@@ -511,6 +511,7 @@ export interface IJupyterConnection extends Disposable {
     readonly url: string;
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     getAuthHeader?(): any; // Snould be a json object
+    readonly workingDirectory?: string;
```

Comment:
Some doc string would be good to explain what this is,

Reply:
@rebornix should this be a Uri? I think that would be better, its more explicit that we're dealing with paths.
Also instead of making this a readonly property woudln't this better be served as a function?

E.g. what if the local machine is Mac and the remote machine is Windows, at this point a simple string replace might not work well, and its best to get a path constructed from the extension author instead .

Reply:
@DonJayamanne it's not exactly a Uri, but more or less a folder/path prefix as we will create at the end is a relative path to the server root. If I understand correctly, when we are creating sessions with a path, we are not really pointing to an absolute path on the server (as the root/pwd in which the server runs is unknown to the client) but we are creating relative paths like `a.ipynb` or `folder/a.ipynb` and the jupyter server will resolve it to a real remote file path on disk.

If the server is started in the synced workspace folder, then `workingDirectory` is not even needed as all we need to do is `resource.fsPath.replace(${workspaceFolder}, '')` (the `workspaceFolder` is local path). However it's possible that a server is started in an ancestor folder of the synced workspace, thus the prefix is needed.

> what if the local machine is Mac and the remote machine is Windows

I'm not sure if the session path is always a normalized unix path (as it's a REST request), we can double check Jupyter's implementation.

Reply:
> I'm not sure if the session path is always a normalized unix path (as it's a REST request), we can double check Jupyter's implementation.

Sorry my concern is the opposite, if the extension author works on windows and the remote ends up being a unix. then extension author mgith always send windows path like `xyz\abc`

I think we'll need to revise this API, feels very much like a kneejerk reaction, i understand you need it for your extension, and lets go with this, and start reviewing the API in a another sprint